pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -88,6 +91,16 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 		if configgen.CanUsePrecomputedCDS(proxy) {
 			if sidecarScope != nil && sidecarScope.XDSOutboundClusters != nil {
 				clusters = append(clusters, sidecarScope.XDSOutboundClusters...)
+				// For locality loadbalancing
+				for i, cluster := range clusters {
+					if cluster.LoadAssignment != nil {
+						// TODO: cache per locality
+						// Currently make a shallow copy of cluster which is fast
+						clone := util.CloneCluster(cluster)
```

Comment:
Any data on this ? Can you add a flag or env variable - so we can measure with it on/off ? Or any benchmark.

Reply:
Can paste some benchmark data

Reply:
It takes less than 1us.
```
root@bz-ubuntu:/mnt/go/src/istio.io/istio/pilot/pkg/networking/util# go test -bench .
goos: linux
goarch: amd64
pkg: istio.io/istio/pilot/pkg/networking/util
BenchmarkCloneCluster-4   	 2000000	       655 ns/op
PASS
ok  	istio.io/istio/pilot/pkg/networking/util	2.037s
```

Reply:
I mean - that's what is limiting the scale pilot can support - the cluster size and changes that result in more garbage than we can collect.